cmd/infra-main/main.go

```diff
@@ -133,8 +134,15 @@ func InitializeAKOInfra() {
 	c.InitNetworkingHandler()
 	lib.RunAviInfraSettingInformer(stopCh)
 	c.AddSecretEventHandler(stopCh)
-	a.SetupSEGroup(transportZone)
-	c.AddAvailabilityZoneCREventHandler(stopCh)
+	segTemplateUUID, err := a.DeriveCloudNameAndSEGroupTmpl(transportZone)
```

Comment:
If there is a failure during SeGroup derivation, we will raise event saying "CloudMatchingTZNotFound", which is not correct.
Better to split this method as well - DeriveCloud and DeriveSEGroup and raise event respectively.
This way the events will be correct according to the error either coming from DeriveCloud or DeriveSEGroup.